src/extension/background-script/actions/nostr/helpers.ts

```diff
@@ -0,0 +1,42 @@
+import * as secp256k1 from "@noble/secp256k1";
+import Hex from "crypto-js/enc-hex";
+import sha256 from "crypto-js/sha256";
+import { Event } from "~/extension/ln/nostr/types";
+
+export function validateEvent(event: Event) {
+  if (event.id !== getEventHash(event)) return false;
+  if (typeof event.content !== "string") return false;
+  if (typeof event.created_at !== "number") return false;
+
+  if (!Array.isArray(event.tags)) return false;
+  for (let i = 0; i < event.tags.length; i++) {
+    const tag = event.tags[i];
+    if (!Array.isArray(tag)) return false;
+    for (let j = 0; j < tag.length; j++) {
+      if (typeof tag[j] === "object") return false;
+    }
+  }
+
+  return true;
+}
+
+export async function signEvent(event: Event, key: string) {
+  return Buffer.from(
+    await secp256k1.schnorr.sign(getEventHash(event), key)
```

Comment:
I am scared of 🐉 with this syntax that uses await in a function parameter.

Reply:
why?

Reply:
maybe because my mind still thinks in polyfills and old JS... and I translate async/await in Promise.resolve chains.
is this reliable in all the different potential environments?

Reply:
the good thing about `await` is that exceptions bubble up naturally as they are not captured in `then()` and `catch()` functions. if this call fails the calling function will capture the exception and return an error to the `nostr.signEvent()` call.

Reply:
I am not saying not to use await. but rather: 
```
const signature =  await secp256k1.schnorr.sign(getEventHash(event), key);
return Buffer.from(signature);
``

(but would be nice to get rid of the Buffer)